lisa/test_example.py

```diff
@@ -38,7 +38,7 @@
 ################################################################################
 
 
-class ExampleTestBundle(RTATestBundle):
+class ExampleTestBundle(RTATestBundle, TestBundle):
```

Comment:
```
class RTATestBundle(FtraceTestBundle, DmesgTestBundle)
class TestBundle(FtraceTestBundle, OptionalDmesgTestBundle, TestBundleBase)
```

So this is only adding the OptionalDmesgTestBundle; how does that work vs DmesgTestBundle? If I get [1] right, then DmesgTestBundle "wins", but in that case what do we gain from the added inheritance?

[1]: https://docs.python.org/3/tutorial/classes.html#multiple-inheritance

Reply:
The goal is to provide a TestBundle class you can safely inherit from and that comes with some convenience features. If a given class inherits explicitly from DmesgTestBundle before TestBundle, it will ensure the non-optional API will "win", so the test can be relied on.

The OptionalDmesgTestBundle will bail out with a warning if it cannot collect the output, and `test_dmesg` will return UNDECIDED with a warning instead of raising an exception if the log file does not exist.